src/main/java/com/microsoft/sqlserver/jdbc/SimpleTtlCache.java

```diff
@@ -0,0 +1,108 @@
+/*
+ * Microsoft JDBC Driver for SQL Server Copyright(c) Microsoft Corporation All rights reserved. This program is made
+ * available under the terms of the MIT License. See the LICENSE file in the project root for more information.
+ */
+
+package com.microsoft.sqlserver.jdbc;
+
+import java.time.Duration;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ThreadFactory;
+import static java.util.concurrent.TimeUnit.SECONDS;
+
+final class SimpleTtlCache<K,V> {    

```

Comment:
add javadocs describing the class and how it's used

Reply:
added